skcosmo/preprocessing/flexible_scaler.py

```diff
@@ -124,78 +153,48 @@ def fit(self, K=None, y=None, K_fit_rows=None, K_fit_all=None):
         :param K_fit_all: an average for the whole kernel matrix
         :type K_fit_all: array
 
-        :return: itself
+        :return: fitted transformer
         """
-        if K is not None:
-            if K.shape[0] != K.shape[1]:
-                raise ValueError(
-                    "The reference kernel is not square, and does not define a RKHS"
-                )
-
-            self.reference_shape_ = K.shape
-
-            if K_fit_rows is not None:
-                if K.shape[0] != len(K_fit_rows):
-                    raise ValueError(
-                        "The supplied column mean does not match the supplied kernel."
-                    )
-            else:
-                K_fit_rows = K.mean(axis=0)
-
-            if K_fit_all is None:
-                K_fit_all = K.mean()
 
+        if K_fit_rows is not None and K_fit_all is not None:
+            self.K_fit_rows_ = K_fit_rows
+            self.K_fit_all_ = K_fit_all
         else:
-            assert K_fit_rows is not None and K_fit_all is not None
-            self.reference_shape_ = [None, len(K_fit_rows)]
+            super().fit(K, y)
 
-        self.K_fit_rows_ = K_fit_rows
-        self.K_fit_all_ = K_fit_all
+        Kc = self._validate_data(K, copy=True, dtype=FLOAT_DTYPES, reset=False)
 
-        Kc = (
-            K
-            - np.broadcast_arrays(K, self.K_fit_rows_)[1]
-            - np.mean(K, axis=1).reshape((K.shape[0], 1))
-            + np.broadcast_arrays(K, self.K_fit_all_)[1]
-        )
+        K_pred_cols = (np.sum(Kc, axis=1) / self.K_fit_rows_.shape[0])[:, np.newaxis]
+
+        Kc -= self.K_fit_rows_
+        Kc -= K_pred_cols
+        Kc += self.K_fit_all_
 
         self.scale_ = np.trace(Kc) / K.shape[0]
 
         return self
 
-    def transform(self, K, y=None):
-        """Centering our Kernel. Previously you should fit data.
+    def transform(self, K, copy=True):
```

Comment:
AFAIK, this is somehow departing from sklearn usual way of doing thing with `transform` first two arguments being X and y. Why remove `y=None` here?